civis/futures.py

```diff
@@ -312,39 +351,96 @@ def __init__(self,
                  max_n_retries=0,
                  client=None,
                  polling_interval=None,
-                 inc_script_names=False):
+                 inc_script_names=False,
+                 n_jobs=-1):
         self.max_n_retries = max_n_retries
         self.hidden = hidden
         self.name = name
+        self.n_jobs = n_jobs
         self.polling_interval = polling_interval
         self.inc_script_names = inc_script_names
         self._script_name_counter = 0
 
         self._shutdown_lock = threading.Lock()
         self._shutdown_thread = False
 
-        if client is None:
-            client = APIClient(resources='all')
-        self.client = client
+        self.client = client or APIClient(resources='all')
 
         # A list of ContainerFuture objects for submitted jobs.
         self._futures = []
-
-    def _make_future(self, job_id, run_id):
-        """Instantiates a :class:`~civis.futures.ContainerFuture`,
-        adds it to the internal list of futures, and returns it.
-        This is a helper method for :func:`submit`.
-        """
-        future = ContainerFuture(job_id, run_id,
-                                 polling_interval=self.polling_interval,
-                                 max_n_retries=self.max_n_retries,
-                                 client=self.client,
-                                 poll_on_creation=False)
-
-        self._futures.append(future)
-
-        # Return a ContainerFuture object with the job ID.
-        return future
+        if self.n_jobs > 0:
+            self._init_worker()
```

Comment:
Should there be a `ValueError` for `n_jobs == 0`?

Reply:
TBH, I am not sure what the conventions are. I can check.

Reply:
So the concurrent.futures convention is that 

1. This parameter is called `max_workers`.
2. If this parameter is `<=0`, it raises a ValueError.
3. If this parameter is None, it defaults to some number computed from the number of cores on your machine. 

joblib conventions are different from these. 

I think the clearest thing is to call this parameter `max_workers` as opposed to `n_jobs` and to use the following.

1. Error on zero
2. Unlimited for `None` or `<0`
3. Limit to max_workers for `>0`.

Reply:
Yeah, prob best to stick to `n_jobs` given joblib.

Reply:
This isn't joblib, though. We can use `max_workers` here and `n_jobs` in the joblib backend.

Reply:
Yep, but then we have to do weird argument name changing elsewhere in the code. :(